core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+    """Class for validating CollectionSnapshotMetadataModel."""
+
+    is_snapshot_metadata_model = True
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return commit_commands_domain.CollectionCommitCmd
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'collection_model': (
                 collection_models.CollectionModel,
-                item.collection_ids),
-            'general_feedback_thread_ids': (
-                feedback_models.GeneralFeedbackThreadModel,
-                item.general_feedback_thread_ids),
-            'creator_ids': (user_models.UserSettingsModel, item.creator_ids),
-            'id': (user_models.UserSettingsModel, [item.id]),
+                [item.id[:item.id.find('-')]]),
+            'committer_model': (
+                user_models.UserSettingsModel, [item.committer_id])
         }
 
+    @classmethod
+    def _validate_collection_model_version_from_item_id(cls, item):
+        """Validate that collection model corresponding to snapshot
+        metadata model has a version greater than or equal to the in item.id.
+
+        Args:
+            item: CollectionSnapshotMetadataModel to validate.
+        """
+        _, collection_model_tuples = cls.external_models['collection_model']
+
+        collection_model = collection_model_tuples[0][1]
+        version = item.id[item.id.rfind('-') + 1:]
+        if int(collection_model.version) < int(version):
+            cls.errors['collection model version check'] = (
+                'Model id %s: Collection model corresponding to '
+                'id %s has a version %s which is less than the version %s in '
+                'snapshot metadata model id' % (
+                    item.id, collection_model.id, collection_model.version,
+                    version))
+
     @classmethod
     def _get_validation_functions(cls):
+        return [cls._validate_collection_model_version_from_item_id]
+
+
+class CollectionSnapshotContentModelValidator(BaseModelValidator):
+    """Class for validating CollectionSnapshotContentModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
         return []
 
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'collection_model': (
+                collection_models.CollectionModel,
+                [item.id[:item.id.find('-')]]),
```

Comment:
Add a comment with the format for item.id here and elsewhere where you are fetching ids for external models by splitting the original model id.